### PR TITLE
Added `envoy` rock

### PR DESCRIPTION
**Pull request by TakoB222**
### Description
---
- Add rockcraft.yaml
- Add tests
- Add tox.ini

### CODE REVIEW

File: envoy/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    plugin: nil
+    source: https://github.com/envoyproxy/envoy.git
+    source-tag: release/v1.27
+    build-packages:
+      - curl
+    build-environment:
+      - ENVOY_VERSION: 1.27.0
+    override-build: |
+      cp $CRAFT_PART_SRC/configs/envoyproxy_io_proxy.yaml $CRAFT_PART_INSTALL
+      curl -L https://github.com/envoyproxy/envoy/releases/download/v${ENVOY_VERSION}/envoy-${ENVOY_VERSION}-linux-x86_64 \
```
Comment (orfeas-k) [inline envoy/rockcraft.yaml:34]: Where does this `curl` command correspond? I don't see any command like this in the upstream [dockerfile](https://github.com/envoyproxy/envoy/blob/main/ci/Dockerfile-envoy#L48)?

File: envoy/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Based on: https://github.com/envoyproxy/envoy/blob/release/v1.27/ci/Dockerfile-envoy
+name: envoy
+summary: Envoy is an open-source edge and service proxy designed for cloud-native applications.
+description: |
+  Envoy is essential for modern applications, particularly those built using microservices, 
+  by providing reliable, efficient, and secure service communication along with valuable observability features.
+version: "1.27.0"
```
Comment (orfeas-k) [inline envoy/rockcraft.yaml:7]: We should use 1.27.3 as upstream [kubeflow pipelines 2.2.0 do](https://github.com/kubeflow/pipelines/blob/2.2.0/third_party/metadata_envoy/Dockerfile#L15)

**Review by orfeas-k — COMMENTED** [review]
Thank you for the PR. I have some comments:
1. I downloaded the rock and exported it to docker with `tox -e export-to-docker` and then ran it with `docker run` and it doesn't run
```bash
╰─$ docker run --rm envoy:1.27.0 -v                                                                                                                                        125 ↵
2024-10-21T14:09:34.760Z [pebble] Started daemon.
2024-10-21T14:09:34.767Z [pebble] POST /v1/services 3.201621ms 202
2024-10-21T14:09:34.770Z [pebble] Service "envoy" starting: /envoy -c /envoyproxy_io_proxy.yaml
2024-10-21T14:09:34.770Z [pebble] Change 1 task (Start service "envoy") failed: cannot start service: fork/exec /envoy: operation not permitted
2024-10-21T14:09:34.777Z [pebble] GET /v1/changes/1/wait 10.388312ms 200
2024-10-21T14:09:34.778Z [pebble] Started default services with change 1.
^C2024-10-21T14:09:36.578Z [pebble] Exiting on interrupt signal.
```

2. Since this is a complex upstream dockerfile image https://github.com/envoyproxy/envoy/blob/v1.27.3/ci/Dockerfile-envoy, would you mind documenting the decisions taken for this rockcraft project? This is a rock that we will need to maintain and if there are changes, we will need to know how this is built in comparison to the upstream oci-image. It's hard to make sense as a reviewer how these 4 lines  in the `envoy` part correspond to this [whole file](https://github.com/envoyproxy/envoy/blob/v1.27.3/ci/Dockerfile-envoy). My initial hunch is that this has to do with how they use it to build the image (e.g. [docker_ci.sh](https://github.com/envoyproxy/envoy/blob/v1.27.3/ci/docker_ci.sh)), but we 'd need this documented (in an issue probably) in order to not forget when we 'll need to update those images in the future.

File: envoy/rockcraft.yaml (hunk 1 above)
Comment (TakoB222) [reply envoy/rockcraft.yaml:34]: Hey @orfeas-k! Yes, but Kimonas mentioned that we don't need build any artefacts here, just copy binary. Here with curl I just copy binary with specific version from release page and run it, isn't it solve this case?

**Pull request closed without merging**